xbmc/video/ContextMenus.cpp

```diff
@@ -228,16 +228,34 @@ class CVideoSelectActionProcessor : public VIDEO::GUILIB::CVideoSelectActionProc
 
 bool CVideoChooseVersion::IsVisible(const CFileItem& item) const
 {
-  return item.HasVideoVersions() || item.HasVideoExtras();
+  return item.HasVideoVersions() && !item.IsVideoAssetNav();
 }
 
 bool CVideoChooseVersion::Execute(const std::shared_ptr<CFileItem>& item) const
 {
   // force selection dialog, regardless of any settings like 'Select default video version'
   item->SetProperty("needs_resolved_video_asset", true);
+  item->SetProperty("video_asset_type", "version");
```

Comment:
Seeing all these string checks you introduced in `CVideoChooser`, have you thought about changing the type of the property to a numeric value? => 

https://github.com/xbmc/xbmc/blob/master/xbmc/video/VideoManagerTypes.h#L19-L24

Reply:
I didn't know if adding a dependency on VideoAssetType here was OK. Easy peasy.